src/components/input_fields/image_upload_field.tsx

```diff
@@ -17,16 +17,19 @@ export interface ImageUploadFieldProps extends InputFieldProps {
     imageShape?: ImageShape;
 
     /** Called when the field changes */
-    onChangeHandler: (id: string, value: ImageFile) => any;
+    onChangeHandler?: (id: string, value: ImageFile) => any;
 
     /** Called when an image file is read */
-    onReadHandler: (id: string, file: ImageFile, contents: string) => any;
+    onReadHandler?: (id: string, dataURL: string) => any;
```

Comment:
This is a breaking change. We could keep the file in the state but the clients using this can also do it if necessary. The file.preview is not necessary, you can also use the dataURL given.

Reply:
I think it's fine, we use this only in cloud front-ends at the moment (in 3? places).